src/main/java/decodex/modules/hex/HexDecoder.java

```diff
@@ -27,8 +30,12 @@ public HexDecoder() {
      * @return new Data object representing the decoded input.
      */
     @Override
-    public Data run(Data data) {
-        String inputString = data.toString().replace(" ", "");
+    public Data run(Data data) throws ModuleException {
+        String inputString = data.toString().toLowerCase(Locale.ROOT);
```

Comment:
Perhaps discard the `Locale.ROOT`? This is because there are a few other string related method calls that also takes in a locale but we mostly didn't specify it, therefore leaving it to Java to use the default.